pkg/ssh/ssh.go

```diff
@@ -163,11 +163,38 @@ func (d *realSSHDialer) Dial(network, addr string, config *ssh.ClientConfig) (*s
 	return ssh.Dial(network, addr, config)
 }
 
+// timeoutDialer wraps an sshDialer with a timeout around Dial().
+type timeoutDialer struct {
+	dialer  sshDialer
+	timeout time.Duration
+}
+
+const sshDialTimeout = 150 * time.Second
+
+var realTimeoutDialer sshDialer = &timeoutDialer{&realSSHDialer{}, sshDialTimeout}
+
+func (d *timeoutDialer) Dial(network, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
+	var client *ssh.Client
+	errCh := make(chan error, 1)
+	go func() {
+		defer runtime.HandleCrash()
+		var err error
+		client, err = d.dialer.Dial(network, addr, config)
```

Comment:
I'm not super familiar with go's barrier model for order of execution across a channel send, but I'd guess it's safe to assume that the client variable gets set before err gets sent on the channel? The alternative would be to pass both the client and err via the channel.